utils/githooks/submodule_update_check.py

```diff
@@ -0,0 +1,51 @@
+"""Adds a warning message to the commit if a module is updated"""
+import re
+import subprocess
+import sys
+
+modified_re = re.compile(r'^(?:M|A)(\s+)(?P<name>.*)')
+
+def rebasing():
+    """Determines if the current operation is a rebase"""
+    with subprocess.Popen(["git", "branch"],
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE) as process:
+
+        return process.communicate()[0].decode().split('\n', maxsplit=1)[0].startswith("* (no branch, rebasing")
+
+def submodule_check(modname, msg_file):
+    """Detemrines if a submodule has been updated"""
+    modified = False
+
+    with subprocess.Popen(['git', 'status', '--porcelain'], stdout=subprocess.PIPE) as p:
+        out, _ = p.communicate()
+        for line in out.decode().splitlines():
+            match = modified_re.match(line)
+            if match:
+                modified = modified | (match.group('name') == modname)
```

Comment:
It's also possible to use the `re.MULTILINE` flag and adjust the expression to avoid needing to split lines and loop at all. We could even create a regex string dynamically that includes `modname`. Then we could simply do something like (untested)
```
re_str = r'^(?:M|A)(\s+)(?P{}.*)'.format(modname)
modified = bool(re.search(re_str, out.decode(), re.MULTILINE))
```
I do have some similar code somewhere else that does something similar if we're interested in fine-tuning